src/fs.rs

```diff
@@ -0,0 +1,453 @@
+//! Access the file system.
+//!
+//! The APIs must be added to `tauri.allowlist.fs` in `tauri.conf.json`:
+//! ```json
+//! {
+//!   "tauri": {
+//!     "allowlist": {
+//!       "fs": {
+//!         "all": true, // enable all FS APIs
+//!         "readFile": true,
+//!         "writeFile": true,
+//!         "readDir": true,
+//!         "copyFile": true,
+//!         "createDir": true,
+//!         "removeDir": true,
+//!         "removeFile": true,
+//!         "renameFile": true,
+//!         "exists": true
+//!       }
+//!     }
+//!   }
+//! }
+//! ```
+//! It is recommended to allowlist only the APIs you use for optimal bundle size and security.
+use js_sys::ArrayBuffer;
+use serde::{Deserialize, Serialize};
+use serde_repr::*;
+use std::path::{Path, PathBuf};
+
+#[derive(Serialize_repr)]
+#[repr(u16)]
+pub enum BaseDirectory {
+    Audio = 1,
+    Cache = 2,
+    Config = 3,
+    Data = 4,
+    LocalData = 5,
+    Desktop = 6,
+    Document = 7,
+    Download = 8,
+    Executable = 9,
+    Font = 10,
+    Home = 11,
+    Picture = 12,
+    Public = 13,
+    Runtime = 14,
+    Template = 15,
+    Video = 16,
+    Resource = 17,
+    App = 18,
+    Log = 19,
+    Temp = 20,
+    AppConfig = 21,
+    AppData = 22,
+    AppLocalData = 23,
+    AppCache = 24,
+    AppLog = 25,
+}
+
+#[derive(Deserialize)]
+pub struct FileEntry {
+    pub path: PathBuf,
+    pub name: Option<String>,
+    pub children: Option<Vec<FileEntry>>,
+}
+
+#[derive(Serialize)]
+pub struct FsDirOptions {
+    pub dir: Option<BaseDirectory>,
+    pub recursive: Option<bool>,
+}
+
+#[derive(Serialize)]
+pub struct FsOptions {
+    pub dir: Option<BaseDirectory>,
+}
+
+#[derive(Serialize)]
+pub struct FsTextFileOption {
+    pub contents: String,
+    path: PathBuf,
+}
+
+pub type BinaryFileContents = ArrayBuffer;
```

Comment:
```suggestion
pub type BinaryFileContents = ArrayBuffer;
```

I think we can just use `ArrayBuffer` directly here

Reply:
I used the `BinaryFileContents` alias because in the `fs` API this maps to
>BinaryFileContents: Iterable<number> | ArrayLike<number> | [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer)

And wasn't sure if there was a way to include the `Iterable<number>` or `ArrayLike<number>` types.